### PR TITLE
Dashboard Cards: Display the 3 latest activities

### LINKED ISSUE #20438: Dashboard Cards: Update dashboard cards endpoint handling
## Tasks

- [ ]  Add new card types parameters
- [ ]  Update request handling for JSON response (parsing, etc)
- [ ]  Update existing dashboard models

### LINKED ISSUE #20447: Dashboard Cards: Display 3 latest activity logs in Activity Log card
## Description

Data can either be from the dashboard cards endpoint or from the activity log endpoint. This is still to be determined.

### LINKED ISSUE #20448: Dashboard Cards: Hide Activity Log card if there's no activity

### LINKED ISSUE #20449: Dashboard Cards: Handle tap actions for the Activity Log card
## Tasks

- [x] Handle tapping an Activity Log cell
- [x] Handle tapping anywhere else in the card

### PR DESCRIPTION
**Pull request by momo-ozawa**
Closes #20447 
Closes #20449
Closes #20448

## Description
- Displays the 3 latest activities in the Activity Log card
- Handles tap actions for the Activity Log card
- Hides the Activity Log card if there's no activities

### Displays the 3 latest activities in the Activity Log card

1. Enable the activity log dashboard card remote feature flag in the debug menu
2. Switch to a site that has activity log enabled and has activities
3. Publish a post
4. Wait for ~1 minute
5. Pull to refresh the dashboard
6. ✅ A "Post modified" activity is displayed in the activity log card

### Handles tap actions for the Activity Log card

1. Enable the activity log dashboard card remote feature flag in the debug menu
2. Switch to a site that has activity log enabled and has activities
3. Tap on "Recent activity" in the activity log card
4. ✅ The activity log screen is displayed
5. Go back to the dashboard
6. Tap on an item in the activity log card
7. ✅ The activity detail screen is displayed

### Hides the Activity Log card if there's no activities

1. Enable the activity log dashboard card remote feature flag in the debug menu
2. Switch to a site that has activity log enabled and has no activities
3. Switch to the menu and tap the activity log to confirm the site doesn't have activities
8. Switch to the dashboard
9. ✅ The activity log card is hidden

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

6. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell+ActivityPresenter.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+// MARK: - JetpackBackupStatusViewControllerDelegate
+
+extension DashboardActivityLogCardCell: JetpackBackupStatusViewControllerDelegate {
+
+    func didFinishViewing() {
+        // Do nothing
+    }
+}
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell+ActivityPresenter.swift]: If we're not doing anything and `JetpackBackupOptionsViewController.backupStatusDelegate` is optional, maybe we can remove this and not set the delegate.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell.swift [hunk 2]
```diff
@@ -80,6 +123,48 @@ final class DashboardActivityLogCardCell: DashboardCollectionViewCell {
...
 }
 
+// MARK: - Diffable DataSource
+
+extension DashboardActivityLogCardCell {
+
+    private func createDataSource() -> DataSource {
+        return DataSource(tableView: tableView) { (tableView, indexPath, activity) -> UITableViewCell? in
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: ActivityTableViewCell.defaultReuseID) as? ActivityTableViewCell else {
+                return UITableViewCell()
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell.swift]: [Nit]:
```suggestion
                return nil
```

**Review by hassaanelgarem — APPROVED** [review]
Works as described 🚀 

I have, however, found one issue where the icon for the video uploaded activity is not displayed. I know this is unrelated to this PR, but there are some activity icons that are not mapped in the app. Looking at `ActivityStyleGuide.stringToGridiconTypeMapping()`, the mapping for the video icon is missing. There might be even more missing icons.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell.swift (hunk 2 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell.swift]: Fixed! [5d20791](https://github.com/wordpress-mobile/WordPress-iOS/pull/20569/commits/5d207910978ef39f3eee77511987b70114df750f)

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell+ActivityPresenter.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Activity Log/DashboardActivityLogCardCell+ActivityPresenter.swift]: Fixed! [240640c](https://github.com/wordpress-mobile/WordPress-iOS/pull/20569/commits/240640c327659a715447e1ff78f2044d876ebb75)

**Comment by momo-ozawa** [comment]
cc: @startuptester 👋 

This PR satisfies the following testing milestones:
- card content and activity types
- context menu
- tap actions
- error handling
- account types and roles
- dark mode
- a11y

**Comment by momo-ozawa** [comment]
@osullivanchris 👋 
Here's what the activity log card looks like on iOS. FYI I reused the cell component from the Activity Log screen and added a date label.

**Comment by osullivanchris** [comment]
> Here's what the activity log card looks like on iOS. FYI I reused the cell component from the Activity Log screen and added a date label.

That's great. I was using the same reference. Glad it was re-usable. 

Everything is looking good to me, nice!

**Pull request merged**